src/Pages/Home/components/AddOperationalCommand.tsx

```diff
@@ -0,0 +1,272 @@
+        </div>
+        <div className="flex flex-col gap-2 pt-2">
+            {operationalTimes.fields.map((field, index) => <div key={field.id} className="flex gap-4 items-center">
+                <div>
+                    start time
+                    <input type="date" {...form.register(`operationalTimes.${index}.start.date`)} />
+                    <input type="time" {...form.register(`operationalTimes.${index}.start.time`)} />
+
+                </div>
+                <div>
+                    end time
+                    <input type="date" {...form.register(`operationalTimes.${index}.end.date`)} />
+                    <input type="time" {...form.register(`operationalTimes.${index}.end.time`)} />
+                </div>
+                {/* TODO: fix the colors */}
+                {index === 0 ? <>
+                    <button type="button" className={`${isPermeate ? ' text-green-700' : 'text-red-600'}`} onClick={() => {
+                    
+                        if (!isPermeate) form.setValue('operationalTimes', [{
+                            ...form.getValues('operationalTimes')[0],
+                            isPermeate: true
+                        }])
+                        else form.setValue(`operationalTimes.${index}.isPermeate`, !isPermeate)
+                        // operationalTimes.update(index, { isPermeate: !isPermeate })
+                    }}>
+                        permeate
+                    </button>
+                </> : <></>}
+                <button type="button"
+                    onClick={() => operationalTimes.remove(index)}>
+                    Remove Time
+                </button>
+            </div>)}
+        </div>
+        <div className="pb-2">
+            {/* FIXME: ask yinon and itamar about the time*/}
+            <button type="button" onClick={() => operationalTimes.append({
+                start: {
+                    date: moment().format(DATE_FORMAT),
+                    time: moment().format(TIME_FORMAT)
+                },
+                end: {
+                    date: moment().add(5, 'hours').format(DATE_FORMAT),
+                    time: moment().add(5, 'hours').format(TIME_FORMAT)
+                },
+                isPermeate: false
+            })}>Add Time</button>
+        </div>
+
+        <button type="button" onClick={() => props.nextStep()}>Next command</button>
+
+    </div>
+
+}
+
+export function CategoryDetails() {
+    const form = useFormContext<AddOperationalCommandFormI>();
+    const technics = form.watch('technics');
+    return <div>
+        <select {...form.register('technicsUsed')} multiple onChange={(e) => {
+            // This is a rare case where we need to cast the selected values to the correct type because the type of the value is string but you know that the value is a TechnicsTypes
+            const selectedValues = Array.from(e.target.selectedOptions, option => option.value) as TechnicsTypes[];
```

Comment:
this is to long function to put in the html 
please put it out and make a reference for the function and use it on the html

Reply:
This comment should be in shake's PR.
I only did a pull to my branch, to update my brach to the main branch, that we agreed that for now it's shaked branch.